glom/core.py

```diff
@@ -753,6 +753,90 @@ def __repr__(self):
         return '%s(%r, args=%r, kwargs=%r)' % (cn, self.func, self.args, self.kwargs)
 
 
+class Partial(object):
+    """
+    makes "upgrading" arbitrary python functions into
+    glom specs more convenient, so that itertools, toolz, et al
+    may be more easily integrated into streaming glom chains
+    """
+    def __init__(self, func, specfunc=None):
+        self.func, self.subspec = func, specfunc
+        self.args = ()
+
+    @classmethod
+    def specfunc(cls, specfunc):
+        return cls(None, specfunc)
+
+    def consts(self, *a, **kw):
+        """pass *a and **kw to func"""
+        ret = Partial(self.func, self.subspec)
+        ret.args = self.args + ('C', a, kw)
+        return ret
+
+    def specs(self, *a, **kw):
+        """glom all of *a and **kw and pass to func"""
+        ret = Partial(self.func, self.subspec)
+        ret.args = self.args + ('S', a, kw)
+        return ret
+
+    def star(self, args=None, kwargs=None):
+        """
+        glom args and kwargs and pass to func as *a and **kw
+        """
+        assert args is not None or kwargs is not None
+        ret = Partial(self.func, self.subspec)
+        ret.args = self.args + ('*', args, kwargs)
+        return ret
+
+    def __repr__(self):
+        chunks = [self.__class__.__name__]
+        if self.func:
+            chunks.append('({!r})'.format(self.func))
+        else:
+            chunks.append('.specfunc({!r})'.format(self.subspec))
+        for i in range(len(self.args) // 3):
+            op, args, kwargs = self.args[i * 3: i * 3 + 3]
+            fname = {'C': 'consts', 'S': 'specs', '*': 'star'}[op]
+            chunks.append('.{}('.format(fname))
+            if op in 'CS':
+                chunks.append(', '.join(
+                    [repr(a) for a in args] +
+                    ['{}={!r}'.format(k, v) for k, v in kwargs.items()]))
+            else:
+                if args:
+                    chunks.append('args=' + repr(args))
+                if args and kwargs:
+                    chunks.append(", ")
+                if kwargs:
+                    chunks.append('kwargs=' + repr(kwargs))
+            chunks.append(')')
+        return ''.join(chunks)
+
+    def __eq__(self, other):
+        return type(self) == type(other) and self.__dict__ == other.__dict__
+
+    def glomit(self, target, scope):
+        all_args = []
+        all_kwargs = {}
+        recurse = lambda spec: scope[glom](target, spec, scope)
+        for i in range(len(self.args) // 3):
+            op, args, kwargs = self.args[i * 3: i * 3 + 3]
```

Comment:
do you think maybe it's better to just hold onto the chunks instead of flattening them? also, https://boltons.readthedocs.io/en/latest/iterutils.html#boltons.iterutils.chunked :)

Reply:
eh 6 one, half-a-dozen the other.... I tend to err on the side of fewer sub-objects running around if it doesn't reduce readability

in this case the code is very simple, and it is 100% an internal concern of the class so I'm not worried about getting it "wrong" -- we can come through later and do profiling or whichever to decide what is better